### ISSUE TITLE
Feature request: argument configuration in package.json

### ISSUE THREAD

**Issue by netsi1964** [yargs/yargs#303]
Love this package but I consider it more appropriate to place the argument configuration for an app in the package.json as a JSON section.

So in there requirements can be defined, not in the actual code.

**Comment by bcoe** [comment]
@netsi1964 this sounds like a good enhancement, I'm not 100% picturing what you're imagining though. Could you share an example `package.json` with the functionality you'd like?

**Comment by netsi1964** [comment]
Hi Benjamin @bcoe ,
I must admit that I am not sure that it is at all a valid thing to do, extend the package.json with _custom_ settings. I have tried to save a package.json with the extended Yargs settings and node has no problem running it. I also tried to add another package with the "save" option, and NPM was able to update the package.json.

If it is you could for instance do like this:
```
{
	"name": "myapp",
	"version": "1.0.0",
	"description": "example",
	"main": "index.js",
	"scripts": {
		"test": "echo \"Error: no test specified\" && exit 1",
		"start": "node app.js"
	},
	"author": "Not Important",
	"license": "ISC",
	"dependencies": {
		"node-persist": "0.0.6",
		"yargs": "^3.15.0"
	},
	"yargs": {
		"commands": [{
			"name": "create",
			"description": "Create a new accound",
			"options": [{
				"name": "name",
				"demand": true,
				"alias": "n",
				"description": "Account name (eg: Twitter, Facebook)",
				"type": "string"
			}, {
				"name": "username",
				"demand": true,
				"alias": "u",
				"description": "Account username or email",
				"type": "string"
			}]
		}, {
			"name": "get",
			"description": "Get an existing account",
			"options": [{
				"name": "name",
				"demand": true,
				"alias": "n",
				"description": "Account name (eg: Twitter, Facebook)",
				"type": "string"
			}]
		}]
	}
}
```
Guess that you recognize and follow the settings I try to describe here.

**Comment by doowb** [comment]
This is something I've thought about too and we've implemented some similar configurations in our command line apps. We created [this map-config](https://github.com/doowb/map-config) library to assist in mapping declaritive configurations to application apis. The [examples](https://github.com/doowb/map-config/blob/master/example) use [a property](https://github.com/doowb/map-config/blob/master/package.json#L30-L38) in the `package.json` as the config and map it to a fake application.

If you're interested in this approach, I'll put together a PR that will allow specifying the configuration in the package.json and mapping it to the api.
